Board.java

```diff
@@ -13,15 +13,29 @@ public String getState() {
         return this.state;
     }
 
+     public Boolean avaliablePos(int pos) {
+        if (state.charAt(pos) == '-') {
+            if (!(pos > 9) || !(pos < 1)) {
+                return true;
+            }
+        }
+        System.out.println("Not a valid position");
+        return false;
+    }
+
     public Board playerMakesMove(char ply, int pos) {
         StringBuilder currentBoard = new StringBuilder(state);
-        currentBoard.setCharAt(pos, ply);
-        Board newBoard = new Board(currentBoard.toString());
-        if (newBoard.checkForWin(ply)) {
-            System.out.print(ply + " won! \n");
-            return new Board("----------");
+        if (avaliablePos(pos)) {
+            currentBoard.setCharAt(pos, ply);
```

Comment:
What I would like to see is this:
`Board newBoard = currentBoard.makeMove(pos, ply)`

And then `Board.makeMove` looks like this:

```
if (isAvailable(position)) {
    String newState = state.toString();
    newState[position] = player;
    return new Board(newState);
 } else {
   return this;
}```